curriculum/challenges/english/20-upcoming-python/learn-string-manipulation-by-building-a-cipher/6552008c0d9d9075cbec9772.md

```diff
@@ -0,0 +1,55 @@
+---
+id: 6552008c0d9d9075cbec9772
+title: Step 12
+challengeType: 20
+dashedName: step-12
+---
+
+# --description--
+
+In this project, you are going to build a cipher. You want to take each letter in your plain message, see the actual position in the alphabet, take the letter located after 3 positions, and replace the original letter with that. This kind of cipher is called *Caesar* cipher, from Julius Caesar.
+
+Let's start by finding the position occupied by the first letter in our string. One way is to use the built-in `find()` function. The syntax is the following:
+
+```py
+a_string.find(char)
+```
+
+In the example above, `char` is the character you want to locate and `a_string` is the string you want to parse.
+
+At the end of your code, call `find()` on your `alphabet` string and pass `text[0]` to the function.
```

Comment:
I might suggest moving the first paragraph to the first step of this project.

````suggestion
In this project, you are going to build a cipher. Specifically, take each letter in your message, see its position in the alphabet, take the letter located after 3 positions, and replace the original letter with that. This kind of cipher is called a *Caesar* cipher.

Start by finding the position occupied by the first letter in the string. One way is to use the built-in `find()` function:

```py
a_string.find(char)
```

Above, `char` is the character you want to locate, and `a_string` is the string you want to parse.

At the end of your code, call `find()` on your `alphabet` string and pass `text[0]` to the function.
````

Reply:
> Should I specify that the project starts with a Caesar cipher and then it gets refactored into a different cipher?

That probably makes sense.

> Is it fine to repeat the same things just after the intro?

I would ere on the side of saying as little as necessary - do not repeat.